### PR TITLE
Users/v pnizetic/places feature flag

### PR DESCRIPTION
**Pull request by nizetic**

### CODE REVIEW

File: .github/workflows/azure-static-web-apps-AZ.yml [hunk 1]
```diff
@@ -24,3 +26,5 @@ jobs:
           app_location: '/' # App source code path relative to repository root
           app_build_command: 'npm run build:test'
           output_location: 'build' # Built app content directory, relative to app_location - optional
+        env: # Add environment variables here
+          REACT_APP_STAGING_ENV: ${{ github.event_name == 'pull_request' && 'true' || 'false' }}
```
Comment (shnooshnoo) [inline .github/workflows/azure-static-web-apps-AZ.yml:30]: why do we need `&& 'true' || 'false'` here?

File: src/common/api/conversions.js [hunk 2]
```diff
@@ -10,6 +12,16 @@ const getUploads = () => {
   return fetch(url);
 };
 
+export const deleteUploads = async id => {
```
Comment (shnooshnoo) [inline src/common/api/conversions.js:15]: /nit async here and below is redundant

File: src/common/languages.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  tt: 'Tatar',
+  tw: 'Twi',
+  ty: 'Tahitian',
+  ug: 'Uighur; Uyghur',
+  uk: 'Ukrainian',
+  ur: 'Urdu',
+  uz: 'Uzbek',
+  ve: 'Venda',
+  vi: 'Vietnamese',
+  vo: 'Volapük',
```
Comment (shnooshnoo) [inline src/common/languages.js:177]: haha why is this here

File: src/common/store/conversion.store.js [hunk 4]
```diff
@@ -209,6 +225,7 @@ export const useConversionStore = create((set, get) => ({
             conversionOperationLog: JSON.stringify(data, null, 4),
             conversionEndTime: Date.now(),
             diagnosticPackageLocation: `${data.properties.diagnosticPackageLocation}&subscription-key=${subscriptionKey}`,
+            // https://us.t-azmaps.azurelbs.com/mapData/84c0efd1-cb13-cb22-6ff1-ebc3ee8b4a9b?api-version=2.0
```
Comment (shnooshnoo) [inline src/common/store/conversion.store.js:228]: is this needed here

File: src/common/store/progress-bar-steps.js [hunk 5]
```diff
@@ -13,28 +15,43 @@ export const progressBarStepsByKey = {
...
-    href: PATHS.REVIEW_CREATE,
-  },
-];
+
+export const useProgressBarSteps = () => {
+  const { isPlacesPreview } = useFeatureFlags();
+
+  const { isRunningIMDFConversion } = useIMDFConversionStatus();
+
+  return useMemo(() => {
```
Comment (shnooshnoo) [inline src/common/store/progress-bar-steps.js:24]: feels like useMemo here is redundant and simple return [...] would work better

File: src/common/store/progress-bar-steps.js [hunk 6]
```diff
@@ -82,27 +99,53 @@ const levelsSelector = s => [
...
+        completedSteps.push(progressBarStepsByKey.layers);
+      }
+    }
+  } else {
+    if (layersPageVisited && allLayersValid(layers)) {
+      completedSteps.push(progressBarStepsByKey.layers);
+    }
   }
+
+  // Facitlity name is required for places preview
```
Comment (shnooshnoo) [inline src/common/store/progress-bar-steps.js:146]: typo

File: src/common/store/review-manifest.store.js [hunk 7]
```diff
@@ -85,17 +85,17 @@ export const useReviewManifestJson = () => {
...
-      .map(layer => {
+      .filter(layer => !layer?.isDraft)
+      .map((layer = {}) => {
         const featureClass = {
           featureClassName: layer.name,
           dwgLayers: layer.value,
         };
 
-        const props = layer.props.filter(prop => !prop.isDraft);
+        const props = (layer.props || []).filter(prop => !prop?.isDraft);
```
Comment (shnooshnoo) [inline src/common/store/review-manifest.store.js:98]: It's hard to comprehend cause this is massive PR but it feels a bit odd that you have to add default values in so many places. Perhaps it's better to validate data once somewhere at the top level, rather than adding default values, checks and optional chainings that often

File: src/components/route/route.js [hunk 8]
```diff
@@ -41,6 +49,9 @@ const Route = ({ title, component: Component, dataRequired }) => {
     }
   }, [dataRequired, lroStatus, navigate]);
 
+  const displayTitle = (isPlacesPreview && overrides.isPlacesPreview.title) || title;
```
Comment (shnooshnoo) [inline src/components/route/route.js:52]: ternary operator fits better here

File: src/pages/georeference/control/controlClass.js [hunk 9]
```diff
@@ -1,13 +1,15 @@
...
 class GeoreferenceControl {
   onAdd(map) {
     this.map = map;
     this.container = document.createElement('div');
     this.container.classList.add('azure-maps-control-container');
-    render(<Control map={this.map} />, this.container);
+
+    const root = createRoot(this.container);
+
+    root.render(<Control map={this.map} />);
```
Comment (shnooshnoo) [inline src/pages/georeference/control/controlClass.js:12]: just curious whats the difference here?

**Review by shnooshnoo — APPROVED** [review]

File: .github/workflows/azure-static-web-apps-AZ.yml (hunk 1 above)
Comment (nizetic) [reply .github/workflows/azure-static-web-apps-AZ.yml:30]: That is a fair point. The solution was derived from more general one:
`'pull_request' && 'staging-value' || 'production-value'`
so I wasn't really thinking about it

File: package.json [hunk 10]
```diff
@@ -8,14 +8,15 @@
     "@fluentui/react-components": "^9.41.0",
     "@turf/turf": "^6.5.0",
     "@zip.js/zip.js": "^2.7.31",
-    "azure-maps-control": "^3.0.2",
-    "azure-maps-indoor": "^0.2.3",
+    "azure-maps-control": "^2.2.7",
```
[the remark this answers is not in the document]
Comment (nizetic) [reply package.json:11]: Yes. There was an issue with the latest version, so I downgraded to make it work, until the issue in the lib is fixed

File: src/common/api/conversions.js (hunk 2 above)
Comment (nizetic) [reply src/common/api/conversions.js:15]: You are right. I will remove it

File: src/common/languages.js (hunk 3 above)
Comment (nizetic) [reply src/common/languages.js:177]: According to Wikipedia, there are 20 speakers of it, we don't want to let them down haha
On a more serious note, this is a full list of [ISO 639-1](https://en.wikipedia.org/wiki/ISO_639-1) language codes.

File: src/common/store/conversion.store.js (hunk 4 above)
Comment (nizetic) [reply src/common/store/conversion.store.js:228]: Most definitely not. Thanks for pointing out!

File: src/common/store/response.store.js [hunk 11]
```diff
@@ -34,7 +34,11 @@ export const useResponseStore = create((set, get) => ({
       errorMessage: '',
     });
   },
-  uploadFile: file => {
+  uploadFile: async (file, { isPlacesPreview }) => {
```
[the remark this answers is not in the document]
Comment (nizetic) [reply src/common/store/response.store.js:37]: Fair point. BTW the return value is never used

File: src/components/route/route.js (hunk 8 above)
Comment (nizetic) [reply src/components/route/route.js:52]: I don't necessarily agree. In case of the flag being true it should return override title (if exists) or normal title as fallback.
I any other case it should just return title. The tertiary counterpart would be:
`isPlacesPreview ? overrides.isPlacesPreview.title || title : title` or 
`isPlacesPreview && overrides.isPlacesPreview.title ?  overrides.isPlacesPreview.title : title`
For me personally, this looks uglier

File: src/pages/georeference/control/controlClass.js (hunk 9 above)
Comment (nizetic) [reply src/pages/georeference/control/controlClass.js:12]: It is the React 18 syntax, previous was React 17, which was generating warnings in the console

File: src/components/route/route.js (hunk 8 above)
Comment (shnooshnoo) [reply src/components/route/route.js:52]: uglier cause its longer? shorter is not equal better.
Here I actually meant isPlacesPreview ? overrides.isPlacesPreview.title : title
I did not see at a glance that important logic here is also to fallback to title when isPlacesPreview.title is falsy.
Unintended usage of operators (&& in this case) may make the code shorter, but it doesn't make it easier to read
